Assets/Scripts3D/Weapons/Weapon.cs

```diff
@@ -4,19 +4,49 @@
 
 public class Weapon : MonoBehaviour
 {
-
     [SerializeField] Camera FPCamera;
     [SerializeField] float Range = 100f;
     [SerializeField] float damage = 20f;
     [SerializeField] ParticleSystem MuzzleFlash;
+    [SerializeField] Transform EquippedWeapon;
+
+    [SerializeField] PlayerController PlayerController;
+
+    private float movementCounter;
```

Comment:
```suggestion
    private float _movementCounter;
```
Private variables should begin with an underscore